plugins/available/goenv.plugin.bash

```diff
@@ -0,0 +1,18 @@
+cite about-plugin
+about-plugin 'load goenv, if you are using it'
+
+# Don't modify the environment if we can't find the tool:
+# - Check if in $PATH already
+# - Check if installed manually to $HOME
+_command_exists goenv ||
+  [[ -f "$HOME/.goenv/bin/goenv" ]] ||
+  return
+
+# Set GOENV_ROOT, if not already set
+export GOENV_ROOT="${GOENV_ROOT:-$HOME/.goenv}"
+
+# Add GOENV_ROOT/bin to PATH, if not already set
+pathmunge "$GOENV_ROOT/bin"
+
+# Initialize goenv, if it hasn't been already
+type goenv 2>/dev/null | grep -q 'is a function' || eval "$(goenv init - bash)"
```

Comment:
Hi @cornfeedhobo thanks for clarifying that goenv may be installed outside of `$HOME/.goenv`, that makes sense.  I think this is coming along, but also don't think the logic is not quite polished yet and have a few more comments:

---

We didn't actually confirm that `$HOME/.goenv` exists before setting `GOENV_ROOT`

If, for some reason, `goenv` were installed by brew and also happened to be installed in `$HOME/.goenv/bin` we couldn't know for sure which the user prefers, but I would argue that the user taking the time to install it via brew indicates a preference ... With that, I think we should either check if the command exists and only pathmunge if not, or use the `after` argument to pathmunge to add it to the end of the path vs the beginning.

it would be odd if this script sets `GOENV_ROOT` (i.e not already set), but DOESN'T execute the eval, since it could result in GOENV_ROOT not matching values defined/expected in the previously-executed eval.  i.e. If we set GOENV_ROOT, we should probably execute the eval.

Q: if a `goenv` function is already defined, does it make sense to assume that goenv is already configured?  i.e maybe check for the function at the top the file and short-circuit out if present?

Side Q: Will the `goenv` function "safely" re-execute the eval if asked?  i.e. if you simply executed the eval twice on the command line, what would happen?

re: `type goenv 2>/dev/null`:
`type` accepts a `-t` argument which returns a single word for the type ( or `""` if not found)  -- Not sure if there version/portability limitations of this flag but I prefer its explicitness over grepping for a string.

Reply:
> Q: if a goenv function is already defined, does it make sense to assume that goenv is already configured? i.e maybe check for the function at the top the file and short-circuit out if present?

A: As seen with all the other plugins, no we should not be bailing. the point is to configure the tool in the same pattern as all the other plugins. if the user wants something else, they shouldn't use this plugin. we should not be bothering ourselves with someone shooting themselves.

> Q: Will the goenv function "safely" re-execute the eval if asked? i.e. if you simply executed the eval twice on the command line, what would happen?

No, but i added a check for that

> re: type goenv 2>/dev/null

Good catch.